pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1052,6 +1059,47 @@ func pruneBundle(tx *gorm.DB, trustDomainID string, expiry time.Time, log logrus
 	return changed, nil
 }
 
+func revokeX509CA(tx *gorm.DB, trustDomainID string, publicKey crypto.PublicKey) error {
+	model := &Bundle{}
+	if err := tx.Find(model, "trust_domain = ?", trustDomainID).Error; err != nil {
+		return sqlError.Wrap(err)
+	}
+
+	bundle, err := modelToBundle(model)
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to unmarshal bundle: %v", err)
```

Comment:
@MarcosDY Why we are returning GRPC errors here?

Reply:
I found some related code was returning GRPCs 
```
	if err := tx.Find(model, "trust_domain = ?", trustDomainID).Error; err != nil {
		return sqlError.Wrap(err)
	}
```
So for consistency I moved all to statuses, but I was not sure if we really wanted that...

Reply:
We are indeed returning GRPC status error in other sqlstore methods as well, I wasn't aware of it.

Reply:
Yes, we should return the status here. Since the `sqlstore` package is better positioned than anyone to know the exact reason of the error, I think that it makes sense to bubble up the status from here rather than having to parse the error and assign a status in an upper layer.